File: hello-world/src/main/java/com/example/helloworld/core/Saying.java
package com.example.helloworld.core;

/**
 * User: barclayadunn
 * Date: 7/26/13
 * Time: 10:33 AM
 */
public class Saying {
    private final long id;
    private final String content;

    public Saying(long id, String content) {
        this.id = id;
        this.content = content;
    }

    public long getId() {
        return id;
    }

    public String getContent() {
        return content;
    }
}

File: dwScala/src/main/java/com/barclayadunn/helloworld/api/Saying.scala
package com.barclayadunn.helloworld.api

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.common.base.MoreObjects
import org.hibernate.validator.constraints.Length

/**
 * Created by barclay.dunn on 8/28/15.
 */
class Saying {
  private var id: Long = 0L
  @Length(max = 3) private var content: String = null

  def this() {
    this()
  }

  def this(id: Long, content: String) {
    this()
    this.id = id
    this.content = content
  }

  @JsonProperty def getId: Long = {
    return id
  }

  @JsonProperty def getContent: String = {
    return content
  }

  override def toString: String = {
    return MoreObjects.toStringHelper(this).add("id", id).add("content", content).toString
  }
}
